### PR TITLE
tests: arch: x86 mem protection check CPU scrubs regs after syscall

### PR DESCRIPTION
**Pull request by maksimmasalski**
To improve Zephyr security, I decided to create a test
to check upon exit of a system call back to the calling thread,
the kernel scrubs CPU registers for sensitive data.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

**Review by LeiW000 — APPROVED** [review]

File: tests/arch/x86/cpu_scrubs_regs/src/main.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		"movq $0xDEADBEEF, %rsi;\n\t"
+		"movq $0xDEADBEEF, %rdi;\n\t"
+		"movq $0xDEADBEEF, %r8;\n\t"
+		"movq $0xDEADBEEF, %r9;\n\t"
+		"movq $0xDEADBEEF, %r10;\n\t"
+		"movq $0xDEADBEEF, %r11;\n\t"
+		);
+#endif
+#endif
+}
```
Comment (mengxianglinx) [inline tests/arch/x86/cpu_scrubs_regs/src/main.c:53]: how about:
#if CONFIG_X86
#ifndef CONFIG_X86_64
#else
#endif //CONFIG_X86_64
#endif //CONFIG_X86

Comment (maksimmasalski) [reply tests/arch/x86/cpu_scrubs_regs/src/main.c:53]: @mengxianglinx I changed that in both places, in void z_impl_test_cpu_write_reg and in void test_syscall_cpu_scrubs_regs

**Review by mengxianglinx — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/arch/x86/cpu_scrubs_regs/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+tests:
+ arch.x86.syscalls.cpu_scrubs_regs:
+    filter: CONFIG_ARCH_HAS_USERSPACE
```
Comment (MaureenHelm) [inline tests/arch/x86/cpu_scrubs_regs/testcase.yaml:3]: Need to filter on x86 arch

**Review by MaureenHelm — DISMISSED** [review]

File: tests/arch/x86/cpu_scrubs_regs/testcase.yaml (hunk 2 above)
Comment (maksimmasalski) [reply tests/arch/x86/cpu_scrubs_regs/testcase.yaml]: @MaureenHelm I added `arch_allow: x86` and removed filter

**Comment by maksimmasalski** [comment]
Now it should be fine to be merged.

**Review by MaureenHelm — APPROVED** [review]

**Comment by maksimmasalski** [comment]
Closes issue https://github.com/zephyrproject-rtos/zephyr/issues/23874

**Pull request merged**